src/data/pequiv.lean

```diff
@@ -0,0 +1,325 @@
+/-
+Copyright (c) 2019 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+
+import data.equiv.basic data.set.lattice tactic.tauto
+
+universes u v w x
+
+/-- A `pequiv` is a partial equivalence, a representation of a bijection between a subset
+  of `α` and a subset of `β` -/
+structure pequiv (α : Type u) (β : Type v) :=
+(to_fun : α → option β)
+(inv_fun : β → option α)
+(inv : ∀ (a : α) (b : β), a ∈ inv_fun b ↔ b ∈ to_fun a)
+
```

Comment:
These will be used often, for example in analysis (manifolds). Maybe add notation `≃.` for these?
```
infixr ` ≃. `:25 := pequiv
```